extensions/clipmenu/src/index.tsx

```diff
@@ -0,0 +1,29 @@
+import { Clipboard, Icon, MenuBarExtra, updateCommandMetadata } from "@raycast/api";
+import { useCachedState } from "@raycast/utils";
+import { iiafe } from "./utils";
+
+export default function Command() {
+  const [history, setHistory] = useCachedState<string[]>("latest-clipboard-histories", [""]);
+
+  iiafe(async () => {
+    const _history = (
+      await Promise.all(
+        Array(6)
+          .fill(0)
+          .map((_, idx) => Clipboard.readText({ offset: idx })),
+      )
+    ).filter(Boolean) as string[];
+
+    updateCommandMetadata({ subtitle: _history[0] });
+    setHistory(_history);
+  });
+
+  return (
+    <MenuBarExtra icon={Icon.Clipboard} title={history[0]}>
+      <MenuBarExtra.Item title="Clipboard History" />
+      {history.map((text, i) => (
+        <MenuBarExtra.Item key={text + i} title={text} onAction={() => Clipboard.paste(text)} icon={Icon.Clipboard} />
+      ))}
+    </MenuBarExtra>
+  );
+}
```

Comment:
```suggestion
import { Clipboard, Icon, MenuBarExtra, updateCommandMetadata } from "@raycast/api";
import { useCachedPromise } from "@raycast/utils";
import { truncateText } from "./utils";

export default function Command() {
  const { isLoading, data } = useCachedPromise(async () => {
    const _history = await Promise.all(
      Array(6)
        .fill(0)
        .map((_, idx) => Clipboard.readText({ offset: idx })),
    );

    updateCommandMetadata({ subtitle: _history[0] });
    return _history.filter(Boolean) as string[];
  });

  return (
    <MenuBarExtra isLoading={isLoading} icon={Icon.Clipboard} title={data ? truncateText(data[0], 50) : ""}>
      <MenuBarExtra.Section title="Clipboard History">
        {data &&
          data.map((text, i) => (
            <MenuBarExtra.Item
              key={text + i}
              title={truncateText(text, 50)}
              onAction={() => Clipboard.paste(text)}
              icon={Icon.Clipboard}
            />
          ))}
      </MenuBarExtra.Section>
    </MenuBarExtra>
  );
}
```

This might be cleaner, also, you can maybe add a preference to set how many items you want to show in the menubar (and maybe add if it should shown first element in menubar too (and length)